puppet/modules/profiles/manifests/repo/deb.pp

```diff
@@ -0,0 +1,24 @@
+# @summary A profile for the debian repo machines
+#
+# @param https
+#   Whether to enable HTTPS. This is typically wanted but can only be enabled
+#   in a 2 pass setup. First Apache needs to run for Letsencrypt to function.
+#   Then Letsencrypt can be enabled. Also useful to turn off in test setups.
+class profiles::repo::deb (
+  Boolean $https = true,
+  String[1] $stable = '3.11',
```

Comment:
this duplicates the data in profiles::web, not sure how to best externalize this information

Reply:
We have the same problem with RPM repos. In think the best solution is to introduce a key in Hiera and then use aliases. For example, here I introduced some generic keys (that don't map to any class) and then use them in other entries:
https://github.com/theforeman/foreman-infra/blob/7b13ffba6ffef521b9871803dfca5ae13aed8e0a/puppet/data/common.yaml#L2-L14

Reply:
https://github.com/theforeman/foreman-infra/pull/2119